File: frontend/src/services/userInfoService.js
import axios from "axios";

const REST_API_URL = 'http://localhost:8080/api/v1/userinfo';

export const updateBasicInfo = (userId, basicInfo) => {
    return axios.post(`${REST_API_URL}/update-basic-info/${userId}`, null, {
        params: {
            basicInfo: basicInfo
        }
    });
};

export const updateAdditionalInfo = (userId, additionalInfo) => {
    return axios.post(`${REST_API_URL}/update-additional-info/${userId}`, null, {
        params: {
            additionalInfo: additionalInfo
        }
    });
};

export const getBasicInfo = (userId) => {
    return axios.get(`${REST_API_URL}/${userId}/basic-info`);
};

export const getAdditionalInfo = (userId) => {
    return axios.get(`${REST_API_URL}/${userId}/additional-info`);
};

File: backend/plortal/src/main/java/com/plortal/plortal/service/UserInfoService.java
package com.plortal.plortal.service;

import com.plortal.plortal.model.entity.User;

import java.util.Optional;

public interface UserInfoService {
    void updateBasicInfo(Long userId, String basicInfo);

    void updateAdditionalInfo(Long userId, String additionalInfo);

    String getBasicInfo(Long userId);

    String getAdditionalInfo(Long userId);
}
